### PR TITLE
Describe avro fields

### PR DESCRIPTION
**Pull request by bonifacyj**
This extension parses an avro schema and derives a logical schema based on it.
It also adds two more fields to topic entry tags: schema name and schema avro type ("record", "array, "enum" etc)

The following types of avro schema are supported:

- Primitive schemas
- Simple records
- Nested records
- Simple unions of records and primitives, or records + records
- Unions with nested records

Find more about avro types in [Apache documentation](http://avro.apache.org/docs/1.8.2/spec.html)

### CODE REVIEW

File: google-datacatalog-kafka-connector/src/google/datacatalog_connectors/kafka/scrape/avro_schema_parser.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+from avro import schema
+from google.datacatalog_connectors.kafka.config.\
+    metadata_constants import MetadataConstants
+
+
+class AvroSchemaParser:
```
Comment (mesmacosta) [inline google-datacatalog-kafka-connector/src/google/datacatalog_connectors/kafka/scrape/avro_schema_parser.py:6]: Can you add the test class this file: `avro_schema_parser_test.py`?

**Review by mesmacosta — CHANGES_REQUESTED** [review]

File: google-datacatalog-kafka-connector/src/google/datacatalog_connectors/kafka/scrape/avro_schema_parser.py (hunk 1 above)
Comment (bonifacyj) [reply google-datacatalog-kafka-connector/src/google/datacatalog_connectors/kafka/scrape/avro_schema_parser.py:6]: Tests for this class are written already, they are in schema_parser_test.py. I have renamed this class but forgot to rename the corresponding test class, that's why names are not the same. I'll refactor and push changes

Comment (bonifacyj) [reply google-datacatalog-kafka-connector/src/google/datacatalog_connectors/kafka/scrape/avro_schema_parser.py:6]: @mesmacosta now tests fail with an error `AttributeError: module 'google.cloud.datacatalog_v1beta1' has no attribute 'enums'".` Does it maybe have to do with the fact that connectors use the different version of Google Cloud API now? I am not sure, what's the most sustainable way to fix it?

Comment (mesmacosta) [reply google-datacatalog-kafka-connector/src/google/datacatalog_connectors/kafka/scrape/avro_schema_parser.py:6]: Please set the fixed version on the `setup.py` file:

the `google-datacatalog-connectors-commons` should be `google-datacatalog-connectors-commons>=0.5.2,<0.6.0`

and `google-datacatalog-connectors-commons-test` should be `google-datacatalog-connectors-commons-test>=0.5.1,<0.6.0`

Versions greater than 0.6.0 use the new version of datacatalog client library.

Comment (mesmacosta) [reply google-datacatalog-kafka-connector/src/google/datacatalog_connectors/kafka/scrape/avro_schema_parser.py:6]: for reference:
[setup.py](https://github.com/GoogleCloudPlatform/datacatalog-connectors-rdbms/blob/rdbms-0.8.2/google-datacatalog-rdbms-connector/setup.py)

Comment (mesmacosta) [reply google-datacatalog-kafka-connector/src/google/datacatalog_connectors/kafka/scrape/avro_schema_parser.py:6]: On a new PR we can upgrade the kafka connector to work with the [2.0.0]( https://pypi.org/project/google-cloud-datacatalog/) version of datacatalog client library.

Comment (bonifacyj) [reply google-datacatalog-kafka-connector/src/google/datacatalog_connectors/kafka/scrape/avro_schema_parser.py:6]: thanks, Marcelo! all done

**Review by mesmacosta — APPROVED** [review]

**Comment by bonifacyj** [comment]
@mesmacosta do you want to merge or do we want @socha23 to also have a look? you decide :)

**Comment by mesmacosta** [comment]
> @mesmacosta do you want to merge or do we want @socha23 to also have a look? you decide :)
LGTM, I will ask @socha23 if he will review this and get back to you.

**Pull request merged**